packages/jest-reporters/src/SummaryReporter.ts

```diff
@@ -54,13 +52,28 @@ const {npm_config_user_agent, npm_lifecycle_event, npm_lifecycle_script} =
 export default class SummaryReporter extends BaseReporter {
   private _estimatedTime: number;
   private readonly _globalConfig: Config.GlobalConfig;
+  private readonly _summaryThreshold: number;
 
   static readonly filename = __filename;
 
-  constructor(globalConfig: Config.GlobalConfig) {
+  constructor(
+    globalConfig: Config.GlobalConfig,
+    options: SummaryReporterOptions = {},
+  ) {
     super();
     this._globalConfig = globalConfig;
     this._estimatedTime = 0;
+    this.validateOptions(options);
+    this._summaryThreshold = options.summaryThreshold || 20;
+  }
+
+  private validateOptions(options: SummaryReporterOptions) {
+    if (
+      options.summaryThreshold &&
+      typeof options.summaryThreshold !== 'number'
+    ) {
+      throw new TypeError('The option summaryThreshold should be a number');
```

Comment:
Would it make sense to check if `summaryThreshold ` is positive integer? Or am I overthinking?

Reply:
Since it wouldn't do any harm to pass a negative value, I think it's fine not to validate it.